webofneeds/won-owner-webapp/src/main/java/won/owner/web/rest/RestUserController.java

```diff
@@ -392,15 +417,49 @@ public ResponseEntity resetPassword(@RequestBody String password) {
     return null;
   }
 
-  private static void myLogoff(HttpServletRequest request, HttpServletResponse response) {
-    CookieClearingLogoutHandler cookieClearingLogoutHandler = new CookieClearingLogoutHandler(
-      AbstractRememberMeServices.SPRING_SECURITY_REMEMBER_ME_COOKIE_KEY);
-    SecurityContextLogoutHandler securityContextLogoutHandler = new SecurityContextLogoutHandler();
-    cookieClearingLogoutHandler.logout(request, response, null);
-    securityContextLogoutHandler.logout(request, response, null);
-  }
 
   public void setRememberMeServices(RememberMeServices rememberMeServices) {
     this.rememberMeServices = rememberMeServices;
   }
+  
+  /**
+	 * Registers the specified user with password and an opional role.
+	 * Assumes values have already been checked for syntactic validity.
+	 * @param email
+	 * @param password
+	 * @param role
+	 * @throws UserAlreadyExistsException
+	 */
+	private void registerUser(String email, String password, String role) throws UserAlreadyExistsException {
+		User user = userRepository.findByUsername(email);
+		if(user != null) {
+			throw new UserAlreadyExistsException();
+		}
+		try {
+	        PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+	        user = new User(email, passwordEncoder.encode(password),role);
+	        user.setEmail(email);
+	        KeystorePasswordHolder keystorePassword = new KeystorePasswordHolder();
+	        //generate a password for the keystore and save it in the database, encrypted with a symmetric key
+	        //derived from the user's password
+	        keystorePassword.setPassword(
+	        		KeystorePasswordUtils.generatePassword(KeystorePasswordUtils.KEYSTORE_PASSWORD_BYTES),
+	        			 	password);
+	        //keystorePassword = keystorePasswordRepository.save(keystorePassword);
+	        //generate the keystore for the user
+			KeystoreHolder keystoreHolder = new KeystoreHolder();
+			try {
+				KeyStore keystore = keystoreHolder.getKeystore(keystorePassword.getPassword(password));
+			} catch (Exception e) {
+				logger.warn("could not create keystore for user " + email);
```

Comment:
exit the method here? return

Reply:
Yes. Better: throw an IllegalStateException or something like that.